From cb8416ddd222b9e6085092bbb8d785d105bd7fd4 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Mar 2023 20:17:00 +0200
Subject: [PATCH] arm64: dts: st: add st,axi-max-burst-len to dma controllers
 on stm32mp251

On stm32mp25x, the DMA3 instances are interconnected to AXI3 bus. AXI3
protocol supports a maximum burst length of 16 beats.
Use st,axi-max-burst-len to dma controllers nodes to limit burst length to
16 beats.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I35e817ed49c08f79a3682cbe8349165b9761f4d4
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 9c1e49b2e1ac..820cfc753b0f 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -178,6 +178,7 @@ hpdma: dma-controller@40400000 {
 				     <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA1>;
 			#dma-cells = <3>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		hpdma2: dma-controller@40410000 {
@@ -201,6 +202,7 @@ hpdma2: dma-controller@40410000 {
 				     <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA2>;
 			#dma-cells = <3>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		hpdma3: dma-controller@40420000 {
@@ -224,6 +226,7 @@ hpdma3: dma-controller@40420000 {
 				     <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA3>;
 			#dma-cells = <3>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		rifsc: bus@42080000 {
-- 
2.39.5

